crates/chronicle-synth/README.md

```diff
@@ -0,0 +1,105 @@
+# Chronicle Synth
+
+Chronicle Domain [Synth](https://www.getsynth.com/) is a program for
+generating Synth schema for Chronicle. It takes a YAML file as input,
+which contains information about the domain's name, attributes,
+entities, activities, agents, and roles, and generates Synth collections
+that can be used to generate the Chronicle operations available to a
+domain's users.
+
+If you just want to generate untyped Chronicle operations with `synth`
+the collections in `synth` are ready to go. Use the
+`./collect` and `./generate` scripts to generate a set of synthetic
+Chronicle operations.
+
+The core functionality of Chronicle Synth is to generate the
+additional Synth schema specific to each individual Chronicle domain.
+
+## Prerequisites
+
+To use Chronicle Synth, you need to have Synth installed on
+your system. If you don't have it already installed, you can follow
+the [instructions](https://www.getsynth.com/docs/getting_started/installation)
+in the Synth documentation to install it.
+
+## Usage
+
+### Loading domain.yaml Chronicle Domain Definitions
+
+To use Chronicle Synth, you must have a YAML file that
+describes your domain. The file should have the following structure:
+
+```yaml
+name: DomainName
+attributes:
+  AttributeName:
+    type: AttributeType
+entities:
+  EntityName:
+    attributes:
+      - AttributeName1
+      - AttributeName2
+activities:
+  ActivityName:
+    attributes:
+      - AttributeName1
+      - AttributeName2
+agents:
+  AgentName:
+    attributes:
+      - AttributeName1
+roles:
+  - RoleName1
+  - RoleName2
+```
+
+Once you have your YAML file, you can generate Synth schema by running
+the following command:
+
+```bash
+chronicle-domain-synth <path-to-yaml-file>
+```
+
+Or, since `<path-to-yaml-file>` defaults to domain.yaml, if your YAML
```

Comment:
It doesn't appear to, I tried `cargo run` from within `crates/chronicle-synth` as the scripts seem designed to run from there too, and it tried to read `crates/chronicle-synth/domain.yaml`.

With `cargo run domain.yaml` I instead get a,
```
Generating schema for domain: Time Recording.
Error: IO(Os { code: 2, kind: NotFound, message: "No such file or directory" })
```
but no idea which file in that case.

Reply:
I should add, it does all seem to work more happily if I run the `cargo` stuff from the root of the repository but the shell scripts from within the crate directory. Though, I can imagine that all that is in anticipation of this being separated into a repository of its own.

Reply:
Yeah, sorry that needs updating from when I built this in a separate private repo. The command that should work from the chronicle root is `cargo run --bin chronicle-synth --`, and it will read by default a `domain.yaml` in the `chronicle-synth` crate root. The `collect` and `generate` scripts need to be run from `crates/synth/`.

Reply:
> I should add, it does all seem to work more happily if I run the `cargo` stuff from the root of the repository but the shell scripts from within the crate directory. Though, I can imagine that all that is in anticipation of this being separated into a repository of its own.

I'd be open to finding a suitable way to get the scripts working via the Rust CLI.

Reply:
Update: I've made it so that running `cargo run --bin chronicle-synth --` automatically calls the `collect` script and guides the user to finding their Synth collections in `crates/chronicle-synth/collections/`. It's kind of up to the user how they want to use Synth to generate data, so the `crates/chronicle-synth/generate` script is there to show __a__ way to get going with generating data from the collections.